astropy/io/fits/src/compressionmodule.c

```diff
@@ -888,7 +899,7 @@ PyObject* compression_compress_hdu(PyObject* self, PyObject* args)
     PyObject* retval = NULL;
     tcolumn* columns = NULL;
 
-    void* outbuf;
+    void* outbuf = NULL;
```

Comment:
Agreed; overall, it seemed best to just keep any individual piece of code as little reliant as possible on the rest being correct...